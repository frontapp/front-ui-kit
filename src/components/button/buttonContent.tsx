import React, {FC} from 'react';
import styled from 'styled-components';

/*
 * Props.
 */

interface ButtonContentProps {
  /** Button content to render. */
  children: React.ReactNode;
}

/*
 * Style.
 */

const StyledButtonContentDiv = styled.div`
  grid-area: content;
`;

/*
 * Component.
 */

export const ButtonContent: FC<ButtonContentProps> = ({children}) => (
  <StyledButtonContentDiv>{children}</StyledButtonContentDiv>
);
