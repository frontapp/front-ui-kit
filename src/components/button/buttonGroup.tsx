import React, {FC} from 'react';
import styled, {css} from 'styled-components';

/*
 * Props.
 */

interface ButtonGroupProps {
  /** How to align the buttons. Default is left. */
  align?: 'left' | 'right';
  /** Buttons to render. */
  children: React.ReactNode;
}

/*
 * Style.
 */

interface StyledButtonGroupDivProps {
  $align: 'left' | 'right';
}

const StyledButtonGroupDiv = styled.div<StyledButtonGroupDivProps>`
  display: flex;
  flex-flow: row;
  gap: 8px;

  ${p => css`
    justify-content: ${p.$align === 'left' ? 'unset' : 'end'};
  `};
`;

/*
 * Component.
 */

export const ButtonGroup: FC<ButtonGroupProps> = ({children, align = 'left'}) => (
  <StyledButtonGroupDiv $align={align}>{children}</StyledButtonGroupDiv>
);
