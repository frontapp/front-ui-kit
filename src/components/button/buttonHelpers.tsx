import React from "react";
import * as ReactIs from "react-is";

/*
 * Helpers.
 */

/**
 * Takes in children for a component and strips out all content besides the first Icon.
 */
export function renderButtonIconContent(children: React.ReactNode) {
  let hasFoundIcon = false;
  return React.Children.toArray(children).map(child => {
    if (typeof child === 'string' || typeof child === 'number')
      return null;
    if (ReactIs.isFragment(child) || !ReactIs.isElement(child))
      return null;

    // Check for an icon and if we find one make sure we only have 1 icon
    // total that we could render. We also need to display any set colors
    // so the button itself can style the color.
    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/consistent-type-assertions
    if ((child.type as any)?.displayName === 'Icon' && !hasFoundIcon) {
      hasFoundIcon = true;
      return React.cloneElement(child, {shouldDisableColor: true});
    }
    return null;
  });
}
