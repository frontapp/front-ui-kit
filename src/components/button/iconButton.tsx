import React, {FC, MouseEventHandler} from 'react';
import styled, {css} from 'styled-components';

import {alphas, greys, palette} from '../../helpers/colorHelpers';
import {renderFirstIconOnly} from '../../helpers/renderHelpers';

/*
 * Props.
 */

interface IconButtonProps {
  /** Children to render. We only support 1 icon for this button and will ignore all other icons. */
  children: React.ReactNode;
  /** Whether we should render the icon in red. */
  isDanger?: boolean;
  /** Whether the button is disabled. If disabled the onClick will not fire. */
  isDisabled?: boolean;
  /** Whether we should force hover state styles. */
  isActive?: boolean;
  /** Called when the user click on the button. */
  onClick: MouseEventHandler;
}

/*
 * Style.
 */

interface StyledIconButtonProps {
  $isDanger?: boolean;
  $isDisabled?: boolean;
  $isActive?: boolean;
}

const StyledIconButton = styled.button<StyledIconButtonProps>`
  background: transparent;
  border: none;
  padding: 7px;
  border-radius: 8px;

  ${p => addIconColorStyles(p.$isDanger, p.$isDisabled, p.$isActive)};
`;

function addIconColorStyles(isDanger?: boolean, isDisabled?: boolean, isActive?: boolean) {
  if (isDisabled)
    return css`
      color: ${greys.shade40};
    `;
  if (isDanger)
    return css`
      color: ${palette.red[isActive ? 'shade50' : 'shade40']};
      background: ${isActive ? alphas.gray20 : 'unset'};

      &:hover {
        color: ${palette.red.shade50};
        background: ${alphas.gray20};
      }
    `;
  return css`
    color: ${greys[isActive ? 'shade80' : 'shade70']};
    background: ${isActive ? alphas.gray20 : 'unset'};

    &:hover {
      color: ${greys.shade80};
      background: ${alphas.gray20};
    }
  `;
}

/*
 * Component.
 */

export const IconButton: FC<IconButtonProps> = props => {
  const {children, isDanger, isDisabled, isActive, onClick} = props;
  return (
    <StyledIconButton $isDanger={isDanger} $isDisabled={isDisabled} $isActive={isActive} onClick={onClick}>
      {renderFirstIconOnly(children)}
    </StyledIconButton>
  );
};
