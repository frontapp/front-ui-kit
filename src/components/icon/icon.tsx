import React, {FC} from 'react';
import styled from 'styled-components';

import CheckmarkCircle from '../../assets/icons/checkmarkCircle.svg';
import Checkmark from '../../assets/icons/checkmarkSelected.svg';
import ChevronDown from '../../assets/icons/chevronDown.svg';
import ChevronLeft from '../../assets/icons/chevronLeft.svg';
import ChevronRight from '../../assets/icons/chevronRight.svg';
import ChevronUp from '../../assets/icons/chevronUp.svg';
import Close from '../../assets/icons/close.svg';
import ExternalLink from '../../assets/icons/externalLink.svg';
import Minus from '../../assets/icons/minus.svg';
import EllipsisHorizontal from '../../assets/icons/moreHorizontal.svg';
import EllipsisVertical from '../../assets/icons/moreVertical.svg';
import Plus from '../../assets/icons/new.svg';
import PlusCircle from '../../assets/icons/plusCircle.svg';
import Search from '../../assets/icons/search.svg';
import {greys} from '../../helpers/colorHelpers';

/*
 * Constants.
 */

const defaultSize = 16;
const defaultColor = greys.shade50;

const importedIcons = {
  CheckmarkCircle,
  Checkmark,
  ChevronDown,
  ChevronLeft,
  ChevronRight,
  ChevronUp,
  Close,
  ExternalLink,
  Minus,
  EllipsisHorizontal,
  EllipsisVertical,
  Plus,
  PlusCircle,
  Search
};
export type IconName = keyof typeof importedIcons;
// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
export const icons = importedIcons as {[name in IconName]: (props: React.SVGAttributes<SVGElement>) => React.ReactElement};

/*
 * Props.
 */

interface IconProps {
  /** The icon to display. */
  name: IconName;
  /** The color of the icon. */
  color?: string;
  /** Controls the width and height of the icon. */
  size?: number;
  /** If set, no color for the icon will be set. This enables inheriting the parents color. */
  shouldDisableColor?: boolean;
}

/*
 * Style.
 */

interface StyledIconDivProps {
  $size: number;
}

const StyledIconDiv = styled.div<StyledIconDivProps>`
  width: ${p => `${p.$size}px`};
  height: ${p => `${p.$size}px`};
`;

/*
 * Component.
 */

export const Icon: FC<IconProps> = props => {
  const {name, size = defaultSize, color = defaultColor, shouldDisableColor} = props;

  // Pull the icon from the list of available icons.
  // The svg file is converted to a function that is called.
  const icon = icons[name];
  if (!icon)
    return null;

  return (
    <StyledIconDiv $size={size}>
      {icon({
        width: size,
        height: size,
        color: shouldDisableColor ? '' : color,
        preserveAspectRatio: "none"
      })}
    </StyledIconDiv>
  );
};
