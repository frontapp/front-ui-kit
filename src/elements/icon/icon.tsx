import React, {FC} from 'react';
import styled from 'styled-components';

import Archive from '../../assets/icons/archive.svg';
import ArchiveFilled from '../../assets/icons/archiveFilled.svg';
import Assign from '../../assets/icons/assign.svg';
import AssignFilled from '../../assets/icons/assignFilled.svg';
import AttachmentArchive from '../../assets/icons/attachmentArchive.svg';
import AttachmentCalendar from '../../assets/icons/attachmentCalendar.svg';
import AttachmentCode from '../../assets/icons/attachmentCode.svg';
import AttachmentExcel from '../../assets/icons/attachmentExcel.svg';
import AttachmentGeneric from '../../assets/icons/attachmentGeneric.svg';
import AttachmentImage from '../../assets/icons/attachmentImage.svg';
import AttachmentMusic from '../../assets/icons/attachmentMusic.svg';
import AttachmentPdf from '../../assets/icons/attachmentPdf.svg';
import AttachmentPowerpoint from '../../assets/icons/attachmentPowerpoint.svg';
import AttachmentVideo from '../../assets/icons/attachmentVideo.svg';
import AttachmentWord from '../../assets/icons/attachmentWord.svg';
import Calendar from '../../assets/icons/calendar.svg';
import CalendarFilled from '../../assets/icons/calendarFilled.svg';
import CheckmarkBox from '../../assets/icons/checkmarkBox.svg';
import CheckmarkCircle from '../../assets/icons/checkmarkCircle.svg';
import CheckmarkCircleEmpty from '../../assets/icons/checkmarkCircleEmpty.svg';
import Checkmark from '../../assets/icons/checkmarkSelected.svg';
import ChevronDown from '../../assets/icons/chevronDown.svg';
import ChevronLeft from '../../assets/icons/chevronLeft.svg';
import ChevronRight from '../../assets/icons/chevronRight.svg';
import ChevronUp from '../../assets/icons/chevronUp.svg';
import Close from '../../assets/icons/close.svg';
import Copy from '../../assets/icons/copy.svg';
import Edit from '../../assets/icons/edit.svg';
import EditFilled from '../../assets/icons/editFilled.svg';
import Export from '../../assets/icons/export.svg';
import ExternalLink from '../../assets/icons/externalLink.svg';
import Gear from '../../assets/icons/gear.svg';
import GearFilled from '../../assets/icons/gearFilled.svg';
import Import from '../../assets/icons/import.svg';
import Info from '../../assets/icons/info.svg';
import InfoFilled from '../../assets/icons/infoFilled.svg';
import Minus from '../../assets/icons/minus.svg';
import EllipsisHorizontal from '../../assets/icons/moreHorizontal.svg';
import EllipsisVertical from '../../assets/icons/moreVertical.svg';
import Plus from '../../assets/icons/new.svg';
import NewContact from '../../assets/icons/newContact.svg';
import Participant from '../../assets/icons/participant.svg';
import ParticipantFilled from '../../assets/icons/participantFilled.svg';
import PlusCircle from '../../assets/icons/plusCircle.svg';
import Preferences from '../../assets/icons/preferences.svg';
import Search from '../../assets/icons/search.svg';
import Star from '../../assets/icons/star.svg';
import StarFilled from '../../assets/icons/starFilled.svg';
import Trash from '../../assets/icons/trash.svg';
import TrashFilled from '../../assets/icons/trashFilled.svg';
import WarningFilled from '../../assets/icons/warningFilled.svg';
import {greys} from '../../helpers/colorHelpers';

/*
 * Constants.
 */

const defaultSize = 16;
const defaultColor = greys.shade50;

const importedIcons = {
  Archive,
  ArchiveFilled,
  Assign,
  AssignFilled,
  AttachmentArchive,
  AttachmentCalendar,
  AttachmentCode,
  AttachmentExcel,
  AttachmentGeneric,
  AttachmentImage,
  AttachmentMusic,
  AttachmentPdf,
  AttachmentPowerpoint,
  AttachmentVideo,
  AttachmentWord,
  Calendar,
  CalendarFilled,
  CheckmarkCircle,
  CheckmarkCircleEmpty,
  CheckmarkBox,
  Checkmark,
  ChevronDown,
  ChevronLeft,
  ChevronRight,
  ChevronUp,
  Close,
  Copy,
  Edit,
  EditFilled,
  EllipsisHorizontal,
  EllipsisVertical,
  Export,
  ExternalLink,
  Gear,
  GearFilled,
  Import,
  Info,
  InfoFilled,
  Minus,
  NewContact,
  Participant,
  ParticipantFilled,
  Plus,
  PlusCircle,
  Preferences,
  Search,
  Star,
  StarFilled,
  Trash,
  TrashFilled,
  WarningFilled
};
export type IconName = keyof typeof importedIcons;
// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
export const icons = importedIcons as {
  [name in IconName]: (props: React.SVGAttributes<SVGElement>) => React.ReactElement;
};

/*
 * Props.
 */

interface IconProps {
  /** The icon to display. */
  name: IconName;
  /** The color of the icon. */
  color?: string;
  /** Controls the width and height of the icon. */
  size?: number;
  /** If set, no color for the icon will be set. This enables inheriting the parents color. */
  shouldDisableColor?: boolean;
}

/*
 * Style.
 */

interface StyledIconDivProps {
  $size: number;
}

const StyledIconDiv = styled.div<StyledIconDivProps>`
  width: ${p => `${p.$size}px`};
  height: ${p => `${p.$size}px`};
`;

/*
 * Component.
 */

export const Icon: FC<IconProps> = props => {
  const {name, size = defaultSize, color = defaultColor, shouldDisableColor} = props;

  // Pull the icon from the list of available icons.
  // The svg file is converted to a function that is called.
  const icon = icons[name];
  if (!icon)
    return null;

  return (
    <StyledIconDiv $size={size}>
      {icon({
        name,
        width: size,
        height: size,
        color: shouldDisableColor ? '' : color,
        preserveAspectRatio: 'none'
      })}
    </StyledIconDiv>
  );
};
