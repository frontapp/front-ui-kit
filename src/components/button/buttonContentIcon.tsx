import React, {FC} from 'react';
import styled, {css} from 'styled-components';

import {renderButtonIconContent} from './buttonHelpers';

/*
 * Props.
 */

interface ButtonContentIconProps {
  /** Position of the icon. If you want an icon in the center, use ButtonContent and Icon. */
  position?: 'left' | 'right';
  /** Children to render. */
  children: React.ReactNode;
}

/*
 * Style.
 */

interface StyledContentIconDivProps {
  $position: 'left' | 'right';
}

const StyledContentIconDiv = styled.div<StyledContentIconDivProps>`
  ${p => {
    if (p.$position === 'left')
      return css`
        grid-area: left-content;
        margin-right: 8px;
      `;
    return css`
      grid-area: right-content;
      margin-left: 8px;
    `;
  }}
  
`;

/*
 * Component.
 */

export const ButtonContentIcon: FC<ButtonContentIconProps> = ({children, position = 'left'}) => (
  <StyledContentIconDiv $position={position}>{renderButtonIconContent(children)}</StyledContentIconDiv>
);
