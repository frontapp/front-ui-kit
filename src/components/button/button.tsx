import React, {FC, MouseEvent, MouseEventHandler} from 'react';
import styled, {css} from 'styled-components';

import {alphas, greys, palette} from '../../helpers/colorHelpers';
import {fonts, fontSizes, fontWeights, VisualSizesEnum} from '../../helpers/fontHelpers';
import {isComponentInChildren, renderChildrenIgnoreSpecifiedComponents, renderChildrenSpecifiedComponents} from '../../helpers/renderHelpers';
import {makeSizeConstants} from '../../helpers/styleHelpers';
import {ButtonContent} from './buttonContent';
import {IconButton} from './iconButton';

/*
 * Constants.
 */

const lineHeights = makeSizeConstants(16, 16, 20);
const verticalPadding = makeSizeConstants(5, 7, 9);
const horizontalPadding = makeSizeConstants(12, 14);
const buttonFontSize = makeSizeConstants(fontSizes.small, fontSizes.medium, fontSizes.large);
const nonButtonContentChildren = [
  'ButtonContentIcon'
];

/*
 * Props.
 */

export type ButtonTypes =
  'primary' |
  'secondary' |
  'tertiary' |
  'primary-danger' |
  'secondary-danger' |
  'icon' |
  'icon-danger';

interface ButtonProps {
  /** The type of button to render. */
  type?: ButtonTypes;
  /** The size of the button. Note EXTRA_LARGE is not supported. */
  size?: VisualSizesEnum;
  /** Whether the button is disabled. If disabled the onClick will not fire. */
  isDisabled?: boolean;
  /** Content to render. If there is no ButtonContent specified, we will automatically wrap it for you. */
  children?: React.ReactNode;
  /** Whether the button is currently active. */
  isActive?: boolean;
  /** Class name to allow custom styling of the button. */
  className?: string;
  /** The color of the icon to be displayed. */
  iconColor?: string;
  /** Called when the user click on the button. */
  onClick?: MouseEventHandler;
}

/*
 * Style.
 */

const StyledButtonWrapperDiv = styled.div`
  display: inline-block;
`;

interface StyledButtonProps {
  $size: VisualSizesEnum;
  $type: Omit<ButtonTypes, 'icon' | 'icon-danger'>;
  $isActive?: boolean;
  $isDisabled?: boolean;
}

const StyledButton = styled.button<StyledButtonProps>`
  display: grid;
  grid-template-areas: "left-content content right-content";
  font-family: ${fonts.system};
  border-radius: 100px;
  box-sizing: border-box;
  font-weight: ${fontWeights.semibold};

  ${p => addButtonSizeStyles(p.$size)};
  ${p => addButtonTypeStyles(p.$type, p.$isDisabled, p.$isActive)};
`;

function addButtonSizeStyles(size: VisualSizesEnum) {
  return css`
    font-size: ${buttonFontSize[size]};
    line-height: ${lineHeights[size]}px;
    padding: ${verticalPadding[size]}px ${horizontalPadding[size]}px;
  `;
}

function addButtonTypeStyles(type: Omit<ButtonTypes, 'icon' | 'icon-danger'>, isDisabled?: boolean, isActive?: boolean) {
  if (isDisabled)
    return css`
      color: ${greys.shade70};
      border: 1px solid transparent;
      box-shadow: 0 1px 3px ${alphas.black10};
      background: ${greys.shade40};

      /* For tertiary we have slightly different styles. */
      ${type === 'tertiary' && css`
        color: ${greys.shade60};
        background: transparent;
        box-shadow: none;
      `}
    `;

  switch (type) {
    case 'primary':
      return css`
        ${addSharedPrimaryStyles()};
        background: ${palette.blue[isActive ? 'shade50' : 'shade40']};

        &:hover {
          background: ${palette.blue.shade50};
        }
      `;
    case 'primary-danger':
      return css`
        ${addSharedPrimaryStyles()};
        background: ${palette.red[isActive ? 'shade50' : 'shade40']};

        &:hover {
          background: ${palette.red.shade50};
        }
      `;
    case 'tertiary':
      return css`
        background: ${isActive ? alphas.gray10 : 'transparent'};
        border: 1px solid transparent;
        color: ${palette.blue.shade40};

        &:hover {
          background: ${alphas.gray10};
        }
      `;
    case 'secondary-danger':
      return css`
        ${addSharedSecondaryStyles(isActive)};
        color: ${palette.red.shade40};
      `;
    case 'secondary':
    default:
      return css`
        ${addSharedSecondaryStyles(isActive)};
        color: ${greys.shade80};
      `;
  }
}

function addSharedPrimaryStyles() {
  return css`
    color: ${greys.white};
    border: 1px solid transparent;
    box-shadow: 0 1px 3px ${alphas.black40};
  `;
}

function addSharedSecondaryStyles(isActive?: boolean) {
  return css`
    background: ${greys[isActive ? 'shade30' : 'white']};
    border: 1px solid ${alphas.black30};
    box-shadow: 0 1px 3px ${alphas.black10};

    &:hover {
      background: ${greys.shade30};
    }
  `;
}

/*
 * Component.
 */

export const Button: FC<ButtonProps> = ({
  type = 'secondary',
  size = VisualSizesEnum.MEDIUM,
  children,
  isDisabled,
  isActive,
  onClick,
  className,
  iconColor
}) => {
  // Wrap the onClick to check if it is disabled or not defined.
  const onButtonClick = (event: MouseEvent) => {
    if (isDisabled || !onClick)
      return;
    onClick(event);
  };

  // Check if we should be rendering an icon.
  if (type === 'icon' || type === 'icon-danger')
    return (
      <IconButton
        isDanger={type === 'icon-danger'}
        className={className}
        isDisabled={isDisabled}
        isActive={isActive}
        onClick={onButtonClick}
        iconColor={iconColor}
      >
        {children}
      </IconButton>
    );
  return (
    <StyledButtonWrapperDiv>
      <StyledButton className={className} $type={type} $size={size} $isDisabled={isDisabled} $isActive={isActive} onClick={onButtonClick}>
        {renderChildrenSpecifiedComponents(children, nonButtonContentChildren)}
        {renderButtonChildren(children)}
      </StyledButton>
    </StyledButtonWrapperDiv>
  );
};

/*
 * Helpers.
 */

/** Renders the children and checks if we need to wrap the passed in children in the ButtonContent. */
function renderButtonChildren(children: React.ReactNode) {
  const shouldWrapInContent = !isComponentInChildren(children, 'ButtonContent');
  if (!shouldWrapInContent)
    return children;
  return <ButtonContent>{renderChildrenIgnoreSpecifiedComponents(children, nonButtonContentChildren)}</ButtonContent>;
}
