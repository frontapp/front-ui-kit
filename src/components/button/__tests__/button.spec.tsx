import {render} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import {VisualSizesEnum} from '../../../helpers/fontHelpers';
import {Icon} from '../../icon/icon';
import {Button} from '../button';
import {ButtonGroup} from '../buttonGroup';

describe('<Button /> Spec', () => {
  describe('Snapshots', () => {
    it('should render button using defaults', () => {
      const view = render(<Button>Button</Button>);
      expect(view.container).toMatchSnapshot();
    });

    it('should render primary button', () => {
      const view = render(<Button type="primary">Button</Button>);
      expect(view.container).toMatchSnapshot();
    });

    it('should render primary-danger button', () => {
      const view = render(<Button type="primary-danger">Button</Button>);
      expect(view.container).toMatchSnapshot();
    });

    it('should render secondary button', () => {
      const view = render(<Button type="secondary">Button</Button>);
      expect(view.container).toMatchSnapshot();
    });

    it('should render secondary-danger button', () => {
      const view = render(<Button type="secondary-danger">Button</Button>);
      expect(view.container).toMatchSnapshot();
    });

    it('should render tertiary button', () => {
      const view = render(<Button type="tertiary">Button</Button>);
      expect(view.container).toMatchSnapshot();
    });

    it('should render different sized button', () => {
      const view = render(
        <ButtonGroup>
          <Button size={VisualSizesEnum.LARGE}>Button</Button>
          <Button size={VisualSizesEnum.MEDIUM}>Button</Button>
          <Button size={VisualSizesEnum.SMALL}>Button</Button>
        </ButtonGroup>
      );
      expect(view.container).toMatchSnapshot();
    });

    it('should render icon button', () => {
      const view = render(
        <Button type="icon">
          <Icon name="Close" />
        </Button>
      );
      expect(view.container).toMatchSnapshot();
    });

    it('should render icon-danger button', () => {
      const view = render(
        <Button type="icon-danger">
          <Icon name="Close" />
        </Button>
      );
      expect(view.container).toMatchSnapshot();
    });
  });

  it('should issue onClick events when clicked', async () => {
    const user = userEvent.setup();
    const onClickMock = jest.fn();
    const view = render(<Button onClick={onClickMock}>Button</Button>);

    await user.click(view.getByRole('button', {name: /Button/i}));

    expect(onClickMock).toBeCalled();
  });

  it('should not issue onClick events when clicked when isDisabled', async () => {
    const user = userEvent.setup();
    const onClickMock = jest.fn();
    const view = render(<Button onClick={onClickMock} isDisabled>Button</Button>);

    await user.click(view.getByRole('button', {name: /Button/i}));

    expect(onClickMock).not.toBeCalled();
  });
});
