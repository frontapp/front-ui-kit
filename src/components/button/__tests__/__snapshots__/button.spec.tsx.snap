// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Spec Snapshots should render button using defaults 1`] = `
.c2 {
  grid-area: content;
}

.c0 {
  display: inline-block;
}

.c1 {
  display: grid;
  grid-template-areas: 'left-content content right-content';
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Helvetica Neue", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  border-radius: 100px;
  box-sizing: border-box;
  font-weight: 600;
  font-size: 14px;
  line-height: 16px;
  padding: 7px 14px;
  background: #ffffff;
  border: 1px solid rgba(0,0,0,0.1);
  box-shadow: 0 1px 3px rgba(0,0,0,0.04);
  color: #3e3e40;
}

.c1:hover {
  background: #e8eaed;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
    >
      <div
        class="c2"
      >
        Button
      </div>
    </button>
  </div>
</div>
`;

exports[`<Button /> Spec Snapshots should render different sized button 1`] = `
.c3 {
  grid-area: content;
}

.c1 {
  display: inline-block;
}

.c2 {
  display: grid;
  grid-template-areas: 'left-content content right-content';
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Helvetica Neue", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  border-radius: 100px;
  box-sizing: border-box;
  font-weight: 600;
  font-size: 16px;
  line-height: 20px;
  padding: 9px 14px;
  background: #ffffff;
  border: 1px solid rgba(0,0,0,0.1);
  box-shadow: 0 1px 3px rgba(0,0,0,0.04);
  color: #3e3e40;
}

.c2:hover {
  background: #e8eaed;
}

.c4 {
  display: grid;
  grid-template-areas: 'left-content content right-content';
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Helvetica Neue", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  border-radius: 100px;
  box-sizing: border-box;
  font-weight: 600;
  font-size: 14px;
  line-height: 16px;
  padding: 7px 14px;
  background: #ffffff;
  border: 1px solid rgba(0,0,0,0.1);
  box-shadow: 0 1px 3px rgba(0,0,0,0.04);
  color: #3e3e40;
}

.c4:hover {
  background: #e8eaed;
}

.c5 {
  display: grid;
  grid-template-areas: 'left-content content right-content';
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Helvetica Neue", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  border-radius: 100px;
  box-sizing: border-box;
  font-weight: 600;
  font-size: 13px;
  line-height: 16px;
  padding: 5px 12px;
  background: #ffffff;
  border: 1px solid rgba(0,0,0,0.1);
  box-shadow: 0 1px 3px rgba(0,0,0,0.04);
  color: #3e3e40;
}

.c5:hover {
  background: #e8eaed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row;
  -ms-flex-flow: row;
  flex-flow: row;
  gap: 8px;
  -webkit-box-pack: unset;
  -webkit-justify-content: unset;
  -ms-flex-pack: unset;
  justify-content: unset;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        class="c2"
      >
        <div
          class="c3"
        >
          Button
        </div>
      </button>
    </div>
    <div
      class="c1"
    >
      <button
        class="c4"
      >
        <div
          class="c3"
        >
          Button
        </div>
      </button>
    </div>
    <div
      class="c1"
    >
      <button
        class="c5"
      >
        <div
          class="c3"
        >
          Button
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`<Button /> Spec Snapshots should render icon button 1`] = `
.c1 {
  width: 16px;
  height: 16px;
}

.c0 {
  background: transparent;
  border: none;
  padding: 7px;
  border-radius: 8px;
  color: #77787a;
  background: unset;
}

.c0:hover {
  color: #3e3e40;
  background: rgba(93,105,133,0.18);
}

<div>
  <button
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        color=""
        height="16"
        name="Close"
        preserveaspectratio="none"
        width="16"
      />
    </div>
  </button>
</div>
`;

exports[`<Button /> Spec Snapshots should render icon-danger button 1`] = `
.c1 {
  width: 16px;
  height: 16px;
}

.c0 {
  background: transparent;
  border: none;
  padding: 7px;
  border-radius: 8px;
  color: #e9483a;
  background: unset;
}

.c0:hover {
  color: #b72215;
  background: rgba(93,105,133,0.18);
}

<div>
  <button
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        color=""
        height="16"
        name="Close"
        preserveaspectratio="none"
        width="16"
      />
    </div>
  </button>
</div>
`;

exports[`<Button /> Spec Snapshots should render primary button 1`] = `
.c2 {
  grid-area: content;
}

.c0 {
  display: inline-block;
}

.c1 {
  display: grid;
  grid-template-areas: 'left-content content right-content';
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Helvetica Neue", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  border-radius: 100px;
  box-sizing: border-box;
  font-weight: 600;
  font-size: 14px;
  line-height: 16px;
  padding: 7px 14px;
  color: #ffffff;
  border: 1px solid transparent;
  box-shadow: 0 1px 3px rgba(0,0,0,0.15);
  background: #367fee;
}

.c1:hover {
  background: #2356b3;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
    >
      <div
        class="c2"
      >
        Button
      </div>
    </button>
  </div>
</div>
`;

exports[`<Button /> Spec Snapshots should render primary-danger button 1`] = `
.c2 {
  grid-area: content;
}

.c0 {
  display: inline-block;
}

.c1 {
  display: grid;
  grid-template-areas: 'left-content content right-content';
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Helvetica Neue", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  border-radius: 100px;
  box-sizing: border-box;
  font-weight: 600;
  font-size: 14px;
  line-height: 16px;
  padding: 7px 14px;
  color: #ffffff;
  border: 1px solid transparent;
  box-shadow: 0 1px 3px rgba(0,0,0,0.15);
  background: #e9483a;
}

.c1:hover {
  background: #b72215;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
    >
      <div
        class="c2"
      >
        Button
      </div>
    </button>
  </div>
</div>
`;

exports[`<Button /> Spec Snapshots should render secondary button 1`] = `
.c2 {
  grid-area: content;
}

.c0 {
  display: inline-block;
}

.c1 {
  display: grid;
  grid-template-areas: 'left-content content right-content';
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Helvetica Neue", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  border-radius: 100px;
  box-sizing: border-box;
  font-weight: 600;
  font-size: 14px;
  line-height: 16px;
  padding: 7px 14px;
  background: #ffffff;
  border: 1px solid rgba(0,0,0,0.1);
  box-shadow: 0 1px 3px rgba(0,0,0,0.04);
  color: #3e3e40;
}

.c1:hover {
  background: #e8eaed;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
    >
      <div
        class="c2"
      >
        Button
      </div>
    </button>
  </div>
</div>
`;

exports[`<Button /> Spec Snapshots should render secondary-danger button 1`] = `
.c2 {
  grid-area: content;
}

.c0 {
  display: inline-block;
}

.c1 {
  display: grid;
  grid-template-areas: 'left-content content right-content';
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Helvetica Neue", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  border-radius: 100px;
  box-sizing: border-box;
  font-weight: 600;
  font-size: 14px;
  line-height: 16px;
  padding: 7px 14px;
  background: #ffffff;
  border: 1px solid rgba(0,0,0,0.1);
  box-shadow: 0 1px 3px rgba(0,0,0,0.04);
  color: #e9483a;
}

.c1:hover {
  background: #e8eaed;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
    >
      <div
        class="c2"
      >
        Button
      </div>
    </button>
  </div>
</div>
`;

exports[`<Button /> Spec Snapshots should render tertiary button 1`] = `
.c2 {
  grid-area: content;
}

.c0 {
  display: inline-block;
}

.c1 {
  display: grid;
  grid-template-areas: 'left-content content right-content';
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Helvetica Neue", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  border-radius: 100px;
  box-sizing: border-box;
  font-weight: 600;
  font-size: 14px;
  line-height: 16px;
  padding: 7px 14px;
  background: transparent;
  border: 1px solid transparent;
  color: #367fee;
}

.c1:hover {
  background: rgba(93,105,133,0.1);
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
    >
      <div
        class="c2"
      >
        Button
      </div>
    </button>
  </div>
</div>
`;
